Sources/LCLPing/ICMP/ICMPPingClient.swift

```diff
@@ -166,6 +174,31 @@ public final class ICMPPingClient: Pingable {
                     }
                     return channel.eventLoop.makeFailedFuture(error)
                 }
+                
+                if let device = self.configuration.device {
+                    switch device.address {
+                    case .v4:
+                        #if canImport(Darwin)
+                        return channel.setOption(.ipOption(.ip_bound_if), value: CInt(device.interfaceIndex))
+                        #elseif canImport(Glibc)
+                        return channel.setOption(.ipOption(.so_bindtodevice), value: device.interfaceIndex)
+                        #endif
+                    case .v6:
+                        #if canImport(Darwin)
+                        return channel.setOption(.ipv6Option(.ip_bound_if), value: CInt(device.interfaceIndex))
+                        #elseif canImport(Glibc)
+                        return channel.setOption(.ipv6Option(.so_bindtodevice), value: device.interfaceIndex)
```

Comment:
on Linux, `so_bind_to_device` is a socket option. We need to use .`socketOption` here